src/db/wallets/walletNonce.ts

```diff
@@ -122,3 +122,36 @@ export const deleteAllNonces = async () => {
     await redis.del(keys);
   }
 };
+
+/**
+ * Resync the nonce i.e., max of transactionCount +1 and lastUsedNonce.
+ * @param chainId
+ * @param walletAddress
+ */
+export const rebaseNonce = async (chainId: number, walletAddress: Address) => {
```

Comment:
Since there's no difference if this func is called more than once, we can put a lock around this and exit if the lock is taken.

Pseudocode:
```
acquire a lock for 10 seconds
if the lock is acquired
  query rpc for transactionCount
  update redis
  release lock
```

Reply:
Actually with a lock, do we even need a lua script?